### PR TITLE
CA-300719: Block export >2TB VDI to VHD format

### PR DESCRIPTION
**Pull request by GeraldElder-Vass**
Successfully blocked >2040GiB vdi export immediately
Did not block exactly 2040GiB vdi export

### CODE REVIEW

**Comment by edwintorok** [comment]
This will also need a pull-request to SM to update the max vhd size there: https://github.com/xapi-project/sm/blob/81a9fcb4094bbb53fb3f9d72cef4cc8499fdbf99/drivers/vhdutil.py#L31
And a similar pull-request to XenRT, because it currently tries to create a VDI of size 2093050 * 1024 * 1024, you'll have to update MAX_VDI_SIZE_LVM and MAX_VDI_SIZE_NFS

**Comment by GeraldElder-Vass** [comment]
Relates to https://github.com/xapi-project/sm/pull/420

File: ocaml/xapi-consts/api_errors.ml [hunk 1]
```diff
@@ -248,6 +248,7 @@ let vdi_in_use = "VDI_IN_USE"
 let vdi_is_sharable = "VDI_IS_SHARABLE"
 let vdi_readonly = "VDI_READONLY"
 let vdi_too_small = "VDI_TOO_SMALL"
+let vdi_too_large = "VDI_TOO_LARGE"
```
Comment (gaborigloi) [inline ocaml/xapi-consts/api_errors.ml]: We also need to defined this error in a second file, like in the original PR: https://github.com/xapi-project/xen-api/pull/3788/files#diff-394b7fa65daaf936d3837fae220fb1b9R709

**Comment by lindig** [comment]
Travis now passes.

**Comment by lindig** [comment]
@gaborigloi is the good to merge?

**Pull request merged**